packages/contracts-bedrock/contracts/universal/CrossDomainMessenger.sol

```diff
@@ -176,6 +176,45 @@ abstract contract CrossDomainMessenger is
      */
     event FailedRelayedMessage(bytes32 indexed msgHash);
 
+    /**
+     * @notice Modifier for a per-message reentrancy guard.
+     */
+    modifier perMessageNonReentrant(bytes32 _msgHash) {
+        bytes32 _hashMsgHash;
+        assembly {
+            // Re-hash the `_msgHash` with the `0xcafebabe` salt to reduce the possibility
+            // of collisions with existing storage slots.
```

Comment:
How does this achieve that?

Reply:
This is salting the message hash, since its yul and not using a mapping would also essentially salts the key before hashing